examples/static-file-server/src/main.rs

```diff
@@ -19,22 +20,31 @@ async fn main() {
         .with(tracing_subscriber::fmt::layer())
         .init();
 
-    let app = Router::new()
-        .nest(
-            "/static",
-            get_service(ServeDir::new(".")).handle_error(|error: std::io::Error| async move {
-                (
-                    StatusCode::INTERNAL_SERVER_ERROR,
-                    format!("Unhandled internal error: {}", error),
-                )
-            }),
+    // `SpaRouter` is the easiest way to serve assets at a nested route like `/assets`
+    // let app = Router::new()
+    //     .route("/foo", get(|| async { "Hi from /foo" }))
+    //     .merge(axum_extra::routing::SpaRouter::new("/assets", "."));
+
+    // for serving assets directly at the root you can use `tower_http::services::ServeDir`
+    // with a `Router` as the fallback
```

Comment:
Why this way around, instead of `ServeDir` as the fallback of a router? (that seems more common to me, that you'd want routes added in code to take preference over files)

Reply:
Something like

```rust
    let app = Router::new()
        .route("/foo", get(|| async { "Hi from /foo" }))
        .fallback(axum::routing::get_service(ServeDir::new(".")).handle_error(...));
```

works too. Maybe thats easier 🤔

Reply:
Yeah it much nicer. You can use `.into_make_service` and `.layer` :)